File: src/main/java/net/zargor/afterlife/web/GroupManagement.java
package net.zargor.afterlife.web;


import com.mongodb.util.JSON;
import net.zargor.afterlife.web.objects.Group;
import org.bson.Document;

import java.util.*;
import java.util.function.*;

/**
 * Created by Zargor on 20.07.2017.
 */
class GroupManagement {

    private List<Group> groups = new ArrayList<>();

    public GroupManagement() {
        List<Document> foundDocs = new ArrayList<>();
        WebServer.getInstance().getMongoDB().getGroupColl().find().forEach((Consumer<? super Document>) foundDocs::add);
        if (!foundDocs.isEmpty()) {
            foundDocs.forEach(d -> groups.add(WebServer.getInstance().getGson().fromJson(d.toJson(), Group.class)));
        }
        //TODO Defaultgruppe in die DB
        groups.add(new Group("default", new ArrayList<>()));
    }

    public synchronized Group getGroup(String name) {
        return groups.stream().filter(g -> Objects.equals(g.getName().toLowerCase(), name.toLowerCase())).findFirst().orElseGet(null);
    }

    public synchronized List<Group> getGroups() {
        return groups;
    }

    public synchronized void createGroup(Group group) {
        if (!groups.contains(group)) {
            groups.add(group);
            WebServer.getInstance().getMongoDB().getGroupColl().insertOne((Document) JSON.parse(WebServer.getInstance().getGson().toJson(group)));
        }
    }

    public synchronized void remGroup(Group group) {
        if (groups.contains(group)) {
            groups.remove(group);
            WebServer.getInstance().getMongoDB().getGroupColl().deleteOne(new Document("name", group.getName()));
        }
    }
}

File: src/main/kotlin/net/zargor/afterlife/web/GroupManagement.kt
package net.zargor.afterlife.web

import com.mongodb.util.JSON
import net.zargor.afterlife.web.objects.Group
import net.zargor.afterlife.web.objects.Session
import org.bson.Document

/**
 * Created by Zargor on 20.07.2017.
 */
class GroupManagement(val main : WebServer) {
    private val groups : MutableList<Group> = mutableListOf()

    init {
        val found : List<Document>? = main.mongoDB.groupColl?.find()?.toList()
        if (found != null) {
            if (!found.isEmpty()) {
                found.forEach { groups.add(main.gson.fromJson(it.toJson(), Group::class.java)) }
            }
        }
    }

    @Synchronized
    fun getGroup(name : String) : Group? = groups.firstOrNull { it.name.toLowerCase() == name.toLowerCase() }

    @Synchronized
    fun getGroups() : MutableList<Group> = groups

    @Synchronized
    fun createGroup(group : Group) {
        if(!groups.contains(group)){
            groups.add(group)
            main.mongoDB.groupColl?.insertOne(JSON.parse(main.gson.toJson(group)) as Document?)
        }
    }
    @Synchronized
    fun deleteGroup(group : Group){
        if(groups.contains(group)){
            groups.remove(group)
            main.mongoDB.groupColl?.deleteOne(Document("name",group.name))
        }
    }
}
